charts/service-binding-operator/templates/tests/test-service-binding-operator-connection.yaml

```
@@ -8,7 +8,11 @@ metadata:
     "helm.sh/hook": test-success
 spec:
   securityContext:
+    {{if eq .Values.kubernetes.platform "openshift"}}
+    runAsNonRoot: true
+    {{- else if or (eq .Values.kubernetes.platform "vanilla") (eq .Values.kubernetes.platform "eks")}}
     runAsNonRoot: false
+    {{- end}}
```

@Kartikey-star Could you please try this?

```suggestion
    runAsNonRoot: !!bool '{{if eq .Values.kubernetes.platform "openshift"}}true{{else}}false{{end}}'
```

---

Chart tested on aws eks and openshift cluster with suggested 
changes.